ext/ADTypesConstructionBaseExt.jl

```diff
@@ -0,0 +1,18 @@
+module ADTypesConstructionBaseExt
+
+using ADTypes: AutoEnzyme, AutoForwardDiff, AutoPolyesterForwardDiff
```

Comment:
Why do you only consider these three types?

Reply:
Why would the other ones need to be specialized?

Reply:
I don't know, as stated below I am not familiar with the packages being discussed here. I just wondered if focusing only on Enzyme and ForwardDiff was a shortcut to answer only @avik-pal's needs or if they are truly the only backends for which such an overload is necessary.

Reply:
It always works with the default constructors, and those are the ones for which it needs to be specialized.

[It seems really weird to say that no one can merge until you review but then say you don't know the packages involved in the extension.....?]

Reply:
https://juliaobjects.github.io/ConstructionBase.jl/dev/#type-tips. others will work without a custom dispatch

Reply:
> [It seems really weird to say that no one can merge until you review but then say you don't know the packages involved in the extension.....?]

On the contrary, it seems rather logical? Due to DI, I am pretty much the only person in the ecosystem using the integrality of ADTypes, so I would appreciate having time to review meaningful changes to this package. I mentioned it because I know you like to merge PRs rather quickly once you approve them (often on a scale of minutes).

The fact that I don't know the packages involved is not a counter argument to my reviewing. On the contrary, I think it's fair that I take some time to understand what happens here. If I added an extension to some SciML package depending on something you'd never heard of, you'd probably want to understand too.

I'll take a look at the link @avik-pal gave, and give a review tomorrow.

Reply:
To Chris's point, this is not going to break anyone's code and is a simple enough change that shouldn't need that long to get merged.

It is blocking a major update to Lux which in turn blocks a correctness update to Flux.

Reply:
To put things in perspective: 24h is a ridiculously short time for something to get merged. Just because Chris often reviews and merges PRs within the hour doesn't mean everyone is held to the same standards.

You'll have a review by me tomorrow, where I make sure that nothing in DI is affected. I endeavour to do the same for all ADTypes PRs. If you don't think my input is necessary, feel free to merge. But I do believe that the special integration of ADTypes and DI would justify that, as a general rule, I am given more than a few minutes to review PRs to this repo. This is especially relevant considering time zone differences (I'm in Europe) and the fact that, unlike Chris, I'm not working full time on SciML.

Reply:
> But I do believe that the special integration of ADTypes and DI would justify that, as a general rule, I am given more than a few minutes to review PRs to this repo. 

I moved the full answer to https://github.com/SciML/ADTypes.jl/issues/95, but in summary, let's make sure you get more tests so you're comfortable here. That's a system problem not a people problem.

But also, if you don't use SetField, why would that break DI? I would presume that if you don't use SetField, then DI doesn't use SetField, so adding SetField support shouldn't even effect DI? I don't see the logic thread here beyond paranoia of missing tests.

>  This is especially relevant considering time zone differences (I'm in Europe) and the fact that, unlike Chris, I'm not working full time on SciML.

? Interesting comment 😅 . For the record, I'm also in Europe (ish) and I have (a few) full time jobs on proprietary codebases that come before any academic work (with open source work part of the academic work). Though I wouldn't recommend

Reply:
> I moved the full answer to https://github.com/SciML/ADTypes.jl/issues/95, but in summary, let's make sure you get more tests so you're comfortable here. That's a system problem not a people problem.

That's a fair answer, I'll react over there. Thank you.

> But also, if you don't use SetField, why would that break DI? I would presume that if you don't use SetField, then DI doesn't use SetField, so adding SetField support shouldn't even effect DI? I don't see the logic thread here beyond paranoia of missing tests.

If it were just an extension indeed there would be zero probability of breaking DI. But as soon as constructors are modified, which is the case here, I think that DI and other packages are more likely to break. The main reason for that is #96.

> For the record, I'm also in Europe (ish) and I have (a few) full time jobs on proprietary codebases that come before any academic work (with open source work part of the academic work).

Point taken, sorry for assuming.

> Though I wouldn't recommend my hours, but that's a topic for another day.

But that's exactly the point, isnt't it? Even from you, no one should expect a review within the hour. You're also entitled to sleep and food and maybe even hobbies. Regardless of the root cause (which we discuss in #95), if 24h doesn't seem like a reasonable reviewing delay, then we have a problem as a community because that really doesn't scale.

Reply:
See #95 for the discussion, no need to continue in two places. TLDR: Chris is right and I should chill a little ^^